src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1462,6 +1418,63 @@ private void ParseModifiers(SyntaxListBuilder tokens)
             }
         }
 
+        /// <summary>
+        /// Return true if the current contextual keyword should be treated as a modifier.
+        /// </summary>
+        private bool IsContextualKeywordActualModifier()
```

Comment:
I feel like the name would be better if it matched your comment.  i.e. "CurrentContextualKeywordShoudlBeTreatedAsAModifier()"